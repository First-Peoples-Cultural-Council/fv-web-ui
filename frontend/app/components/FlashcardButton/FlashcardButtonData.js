import PropTypes from 'prop-types'
import useNavigationHelpers from 'common/useNavigationHelpers'
/**
 * @summary FlashcardButtonData
 * @version 1.0.1
 * @component
 *
 * @param {object} props
 * @param {function} props.children
 *
 */
function FlashcardButtonData({ children }) {
  const { getSearchAsObject, convertObjToUrlQuery, navigate } = useNavigationHelpers()
  const { flashcard: queryFlashcard, ...restQueries } = getSearchAsObject()
  const onClickDisable = () => {
    navigate(`${window.location.pathname}?${convertObjToUrlQuery(Object.assign({}, restQueries))}`)
  }
  const onClickEnable = () => {
    navigate(`${window.location.pathname}?${convertObjToUrlQuery(Object.assign({}, restQueries, { flashcard: true }))}`)
  }
  return children({
    queryFlashcard,
    onClickDisable,
    onClickEnable,
  })
}
// PROPTYPES
const { func } = PropTypes
FlashcardButtonData.propTypes = {
  children: func,
}

export default FlashcardButtonData
