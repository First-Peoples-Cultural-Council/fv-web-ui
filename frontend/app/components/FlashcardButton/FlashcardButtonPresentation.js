import React from 'react'
import PropTypes from 'prop-types'
import FVButton from 'components/FVButton'

/**
 * @summary FlashcardButtonPresentation
 * @version 1.0.1
 * @component
 *
 * @param {object} props
 *
 * @returns {node} jsx markup
 */
function FlashcardButtonPresentation({ queryFlashcard, onClickDisable, onClickEnable }) {
  return queryFlashcard ? (
    <FVButton variant="contained" color="primary" className="WordsList__viewModeButton" onClick={onClickDisable}>
      Cancel flashcard view
    </FVButton>
  ) : (
    <FVButton variant="contained" className="WordsList__viewModeButton" onClick={onClickEnable}>
      Flashcard view
    </FVButton>
  )
}
// PROPTYPES
const { func, oneOfType, string, number } = PropTypes
FlashcardButtonPresentation.propTypes = {
  queryFlashcard: oneOfType([string, number]),
  onClickDisable: func,
  onClickEnable: func,
}
FlashcardButtonPresentation.defaultProps = {
  onClickDisable: () => {},
  onClickEnable: () => {},
}

export default FlashcardButtonPresentation
