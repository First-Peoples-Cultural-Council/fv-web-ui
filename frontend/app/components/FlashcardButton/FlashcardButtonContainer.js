import React from 'react'
import FlashcardButtonPresentation from 'components/FlashcardButton/FlashcardButtonPresentation'
import FlashcardButtonData from 'components/FlashcardButton/FlashcardButtonData'

/**
 * @summary FlashcardButtonContainer
 * @version 1.0.1
 * @component
 *
 * @param {object} props
 *
 * @returns {node} jsx markup
 */
function FlashcardButtonContainer() {
  return (
    <FlashcardButtonData>
      {({ queryFlashcard, onClickDisable, onClickEnable }) => {
        return (
          <FlashcardButtonPresentation
            queryFlashcard={queryFlashcard}
            onClickDisable={onClickDisable}
            onClickEnable={onClickEnable}
          />
        )
      }}
    </FlashcardButtonData>
  )
}

export default FlashcardButtonContainer
