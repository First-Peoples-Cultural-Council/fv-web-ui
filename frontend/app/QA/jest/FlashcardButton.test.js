// Mock: react-redux
// --------------------------------------------------
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}))

// Mock: @material-ui/core/styles
// --------------------------------------------------
jest.mock('@material-ui/core/styles', () => ({
  /* eslint-disable-next-line */
  withStyles: (styles) => (component) => component,
}))

// Let us begin!
// ==================================================
import React from 'react'
import renderer from 'react-test-renderer'
import FlashcardButtonContainer from 'components/FlashcardButton/FlashcardButtonContainer'
import FlashcardButtonData from 'components/FlashcardButton/FlashcardButtonData'
import FlashcardButtonPresentation from 'components/FlashcardButton/FlashcardButtonPresentation'

describe('FlashcardButton', () => {
  // Container
  test('Container', () => {
    const component = renderer.create(<FlashcardButtonContainer />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  // Data
  test('Data', () => {
    // Updated in FlashcardButtonData.children
    let output = {}

    renderer.create(
      <FlashcardButtonData>
        {(dataLayerOutput) => {
          output = dataLayerOutput
          return null
        }}
      </FlashcardButtonData>
    )
    expect(output).toMatchInlineSnapshot(`
      Object {
        "onClickDisable": [Function],
        "onClickEnable": [Function],
        "queryFlashcard": undefined,
      }
    `)
  })
  // Presentation
  test('Presentation ACTIVE flashcard mode', () => {
    const component = renderer.create(<FlashcardButtonPresentation />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Presentation INACTIVE flashcard mode', () => {
    const component = renderer.create(<FlashcardButtonPresentation queryFlashcard={1} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
