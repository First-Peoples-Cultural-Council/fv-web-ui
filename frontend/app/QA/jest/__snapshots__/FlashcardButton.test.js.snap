// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashcardButton Container 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained WordsList__viewModeButton"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Flashcard view
  </span>
</button>
`;

exports[`FlashcardButton Presentation ACTIVE flashcard mode 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained WordsList__viewModeButton"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Flashcard view
  </span>
</button>
`;

exports[`FlashcardButton Presentation INACTIVE flashcard mode 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained WordsList__viewModeButton MuiButton-containedPrimary"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Cancel flashcard view
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
